### ISSUE TITLE
Port remaining IEnumerable extensions from Core

### ISSUE THREAD

**Issue by brandongregoryscott** [AndcultureCode/AndcultureCode.CSharp.Extensions#27]
Looks like there's a few IEnumerable extension methods that didn't make it here yet.

https://github.com/AndcultureCode/AndcultureCode.CSharp.Core/blob/master/src/AndcultureCode.CSharp.Core/Extensions/EnumerableExtensions.cs
```csharp
public static string Join(this IEnumerable<string> list, string delimiter = ", ");
public static string Join(this IEnumerable<KeyValuePair<string, string>> list, string keyValueDelimiter, string delimiter = ", ");
public static string Join(this List<string> list, string delimiter = ", ");
public static string Join(this KeyValuePair<string, string> pair, string delimiter);
```

Port these over and ensure there's coverage added (doesn't seem to be any tests for these)

**Comment by saidshahdev** [comment]
@brandongregoryscott When we port over these methods to the Extensions project should they also be removed from the Core project at the same time?

**Comment by brandongregoryscott** [comment]
@SaidShah Yes, we'd want to remove them from core after they're ported over here. I believe AndcultureCode/AndcultureCode.CSharp.Core#18 can also be closed out after they are ported here + tested

**Comment by brandongregoryscott** [comment]
Published with [v0.4.5](https://github.com/AndcultureCode/AndcultureCode.CSharp.Extensions/releases/tag/v0.4.5)

@saidshahdev If you'd like, you can take on AndcultureCode/AndcultureCode.CSharp.Core#18 as a follow-up to remove the extension methods from core.
